### PR TITLE
community: Batching added in embed_documents of HuggingFaceInferenceAPIEmbeddings

### PR DESCRIPTION
**Pull request by abhishek9998**
- **Description:** Batching added in embed_documents of HuggingFaceInferenceAPIEmbeddings
  - **Issue:** https://github.com/langchain-ai/langchain/issues/16443
  - **Dependencies:** tqdm
  - **Twitter handle:** @Abhishingadiya

### CODE REVIEW

File: libs/community/langchain_community/embeddings/huggingface.py [hunk 1]
```diff
@@ -321,15 +340,30 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
...
-            headers=self._headers,
-            json={
-                "inputs": texts,
-                "options": {"wait_for_model": True, "use_cache": True},
-            },
-        )
-        return response.json()
+        all_embeddings = []
+        length_sorted_idx = np.argsort([-self._text_length(sen) for sen in texts])
+        sentences_sorted = [texts[idx] for idx in length_sorted_idx]
```
Comment (hwchase17) [inline libs/community/langchain_community/embeddings/huggingface.py:345]: where is this used?

Comment (abhishek9998) [reply libs/community/langchain_community/embeddings/huggingface.py:345]: ### Example Code

```
embeddings = HuggingFaceInferenceAPIEmbeddings(
    api_key=inference_api_key, 
    api_url=api_url,
    model_name="bge-large-en-v1.5"
)
pinecone.init(api_key=os.getenv("PINECONE_API_KEY"), environment=environment)

loader = PyPDFDirectoryLoader("data")
docs = loader.load()
text_splitter = RecursiveCharacterTextSplitter(chunk_size=2000, chunk_overlap=200)
chunks = text_splitter.split_documents(docs)

vectordb = Pinecone.from_documents(chunks, embeddings, index_name=index_name, namespace=namespace)
```
this code snippet is getting 314 request code from huggingface.py
```
response = requests.post(
            self._api_url,
            headers=self._headers,
            json={
                "inputs": texts,
                "options": {"wait_for_model": True, "use_cache": True},
            },
        )
        return response.json()
```
we should support batch size here. like local model embedding 

### Description

I am trying to use pinecone with hugging face inference for the embedding model. My total chunks are 420. and it is trying to process in one request.
Also embedding_chunk_size is not parsable from Pinecone.from_documents() method

**Comment by abhishek9998** [comment]
Any Update @baskaryan ?
PS: I am using custom branch of langchain for my prod sue to HuggingFaceInferenceAPIEmbeddings is not supporting batching mechanism.
same issue: https://discuss.huggingface.co/t/batch-size-limit-32/82471

**Comment by efriis** [comment]
closing because this is against the deprecated community integration for HF - could you confirm it's still an issue with the partner package, and submit an issue/pr against langchain-huggingface if so?

**Pull request closed without merging**